### PR TITLE
getsockname mirrors UNIX getsockname now. Added getpeername.

### PR DESCRIPTION
**Pull request by amitmurthy**
Closes https://github.com/JuliaLang/julia/issues/21801

NEWS.md entry will have to wait till 0.6 is released.

### CODE REVIEW

**Comment by tkelman** [comment]
> NEWS.md entry will have to wait till 0.6 is released.

Huh? Several PR's are open that are starting a placeholder for 0.7 NEWS at the top of the file.

File: test/socket.jl [hunk 1]
```diff
@@ -191,56 +191,35 @@ if !is_windows() || Sys.windows_version() >= Sys.WINDOWS_VISTA_VER
 end
 
 begin
-    default_port = UInt16(11011)
-    default_addr = IPv4("127.0.0.1")
+    for (addr, porthint) in [(IPv4("127.0.0.1"), UInt16(11011)),
+                        (IPv6("::1"), UInt16(11012)), (getipaddr(), UInt16(11013))]
+            port, listen_sock = listenany(addr, porthint)
```
Comment (tkelman) [inline test/socket.jl]: excess indent in this block

**Review by vtjnash — APPROVED** [review]

**Comment by amitmurthy** [comment]
Will merge after a green CI.

File: NEWS.md [hunk 2]
```diff
@@ -3,6 +3,8 @@ Julia v0.7.0 Release Notes
 
 New language features
 ---------------------
+  * `getpeername` on a `TCPSocket` returns the address and port of the remote
+    endpoint of the TCP connection ([#21825]).
```
Comment (tkelman) [inline NEWS.md]: this is a library improvement, not a language feature

Comment (amitmurthy) [inline NEWS.md]: ref : https://github.com/JuliaLang/julia/pull/22141

**Pull request merged**